packages/amplify_auth_cognito/ios/Classes/FlutterFetchCognitoSessionResult.swift

```diff
@@ -30,70 +31,147 @@ struct FlutterFetchCognitoSessionResult {
       do {
         id = try identityProvider.getIdentityId().get()
       } catch {
-         throw error as! AuthError
+        throw error as! AuthError
       }
     }
     return id
   }
 
-  func getCredentials(session: AuthSession) throws -> [String: String] {
+func getCredentials(session: AuthSession) throws -> [String: String] {
     var credentialMap: [String: String] = [:]
-
+    let userPoolOnly = isUPOnly(session: session)
+    if (userPoolOnly) {
+        return credentialMap
+    }
     if let awsCredentialsProvider = session as? AuthAWSCredentialsProvider {
-      let creds =  try awsCredentialsProvider.getAWSCredentials().get()
+      do {
+        let creds =  try awsCredentialsProvider.getAWSCredentials().get()
         credentialMap["awsAccessKey"] = creds.accessKey
         credentialMap["awsSecretKey"] = creds.secretKey
+      } catch {
+        throw error as! AuthError
+      }
+      do {
+        let tempCreds = try awsCredentialsProvider.getAWSCredentials().get() as? AuthAWSTemporaryCredentials
+          if (tempCreds?.sessionKey != nil) {
+            credentialMap["sessionToken"] = tempCreds?.sessionKey
+          }
+      } catch {
+        print("Unable to cast session token.")
+      }
     }
-    
     return credentialMap;
   }
 
   func getTokens(session: AuthSession) throws -> [String: String] {
     var tokenMap: [String: String] = [:]
 
     if let cognitoTokenProvider = session as? AuthCognitoTokensProvider {
-      let tokens = try cognitoTokenProvider.getCognitoTokens().get()
-        tokenMap["accessToken"] = tokens.accessToken
-        tokenMap["idToken"] = tokens.idToken
-        tokenMap["refreshToken"] = tokens.refreshToken
+        do {
+            let tokens = try cognitoTokenProvider.getCognitoTokens().get()
+            tokenMap["accessToken"] = tokens.accessToken
+            tokenMap["idToken"] = tokens.idToken
+            tokenMap["refreshToken"] = tokens.refreshToken
+        } catch {
+            tokenMap["error"] = "You are currently signed out."
+        }
     }
     return tokenMap;
   }
+
+  // Checks if the auth configuration includes User Pools only, inferred from the identity result
+  func isUPOnly(session: AuthSession) -> Bool {
+    var isUserPoolOnly = false;
+    do {
+      let identityProvider = session as! AuthCognitoIdentityProvider
+      try identityProvider.getIdentityId().get()
+    } catch {
+      if case .service(_, _, let errorType) = error as? AuthError {
+        if case .invalidAccountTypeException = errorType as? AWSCognitoAuthError {
+          isUserPoolOnly = true
+        }
+      }
+    }
+    return isUserPoolOnly
+  }
+
+  // Checks if the user is a guest via Identity Pool unauthenticated access
+  func isGuest(session: AuthSession) -> Bool {
+    var signedOut: Bool;
+    var credentialsExist: Bool
+    do {
+      let idp = session as! AuthCognitoIdentityProvider
+      try idp.getUserSub().get()
```

Comment:
User might have signed in through the federatedSignIn() method in escape hatch to CIDP. In which case usersub will return an error. But the user is signed In.